v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -0,0 +1,85 @@
+# `log_context_property_basic_types` requirements
+
+`log_context_property_basic_types` is a collection of log context property interface concrete implementations that implement the code needed for handling basic types (integers, bool, etc.).
+
+The list of supported basic types is:
+
+- int64_t
+- uint64_t
+- int32_t
+- uint32_t
+- int16_t
+- uint16_t
+- int8_t
+- uint8_t
+
+Note: More types will be added as needed.
+
+## Exposed API
+
+```
+#define SUPPORTED_BASIC_TYPES \
+    int64_t, \
+    uint64_t, \
+    int32_t, \
+    uint32_t, \
+    int16_t, \
+    uint16_t, \
+    int8_t, \
+    uint8_t
+
+#define DECLARE_BASIC_TYPE(type_name) \
+    extern const LOG_CONTEXT_PROPERTY_TYPE_IF MU_C2(type_name, _log_context_property_type);
+
+MU_FOR_EACH_1(DECLARE_BASIC_TYPE, SUPPORTED_BASIC_TYPES)
+```
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
```

Comment:
>size_t buffer_length [](http://example.com/codeflow?start=87&length=20)

Can this be inferred from type for basic types?

Reply:
Not really. This is the size of the buffer available to write the stringified version.